src/Event.cpp

```diff
@@ -6,35 +7,54 @@
 #include "HGCalSimulation/FastShower/interface/Event.h"
 #endif
 
+Event::Event(uint32_t run, uint32_t event): run_(run), event_(event) {
+}
+
+void Event::fillHit(uint32_t id, double energy) {
+    auto itr_insert = hits_.emplace(id, energy);
+    // if id already inserted (noise and hit), add the energy
+    if(!itr_insert.second) itr_insert.first->second += energy;
+}
+
+void Event::fillGenEn(uint32_t id, double energy) {
+    generated_energy_.emplace(id, energy);
+}
+
+void Event::fillGenEta(uint32_t id, double eta) {
+    generated_eta_.emplace(id, eta);
+}
+
+void Event::fillGenPhi(uint32_t id, double phi) {
+    generated_phi_.emplace(id, phi);
+}
+
+void Event::fillPDGid(uint32_t id, int PDGid) {
+    pdgid_.emplace(id, PDGid);
+}
+
+void Event::fillThick(uint32_t id, int thick) {
+    if (!(thick_.find(id) != thick_.end()))
+        thick_.emplace(id, thick);
```

Comment:
Safer to use one single line or brackets for the `if` statement

Reply:
Fixed